### PR TITLE
Consolidate message type errors

### LINKED ISSUE #309: Consolidate message type errors
The following pattern appears a lot in the code, especially in the `from_message` functions:

```rust
if message.message_type() != MessageType::Auxinfo(AuxinfoMessageType::R1CommitHash) {
        error!(
            "Encountered unexpected MessageType. Expected {:?}, Got {:?}",
            MessageType::Auxinfo(AuxinfoMessageType::R1CommitHash),
            message.message_type()
        );
        return Err(InternalError::InternalInvariantFailed);
}
```

We should consolidate this into a function.

## Proposed solutions

The function could be on the `Message` type:

```rust
fn check_type(&self, expected_type: MessageType) -> Result<()> { ... }
```
Maybe also update the logged message to include something about "A message was misrouted.".

## Closing criteria
- [ ] Replace all instances of the pattern with a function

### PR DESCRIPTION
**Pull request by hridambasu**
Closes #309

### CODE REVIEW

File: src/broadcast/data.rs
Comment (marsella) [inline src/broadcast/data.rs:1]: todo: remove commented-out code in this file.

File: src/messages.rs [hunk 1]
```diff
@@ -151,4 +154,17 @@ impl Message {
     pub fn to(&self) -> ParticipantIdentifier {
         self.to
     }
+
+    /// Check message type
```
Comment (marsella) [inline src/messages.rs:158]: todo: Our doc comments conventions require full sentences with ending punctuation. Please add a `.` here.

File: src/messages.rs [hunk 2]
```diff
@@ -151,4 +154,17 @@ impl Message {
...
+    pub fn check_type(&self, expected_type: MessageType) -> Result<()> {
```
Comment (marsella) [inline src/messages.rs:159]: todo: I don't think this function needs to be public. `pub(crate)` is fine.

**Review by marsella — CHANGES_REQUESTED** [review]
Thanks! Three little notes.

**Comment by hridambasu** [comment]
Thanks for the comments @marsella! All of them are addressed. Ready for re-review!

**Review by marsella — APPROVED** [review]
Looks good, thank you!

**Pull request merged**